packages/react-components/react-avatar/src/components/AvatarGroupItem/AvatarGroupItem.test.tsx

```diff
@@ -1,25 +1,53 @@
 import * as React from 'react';
-import { render } from '@testing-library/react';
+import { render, screen } from '@testing-library/react';
 import { AvatarGroupItem } from './AvatarGroupItem';
+import { AvatarGroupContext } from '../../contexts';
 import { isConformant } from '../../common/isConformant';
 
+const testId = 'testId';
+
+const ContextWrapper: React.FC = ({ children }) => (
+  <AvatarGroupContext.Provider value={{ isOverflow: true }}>{children}</AvatarGroupContext.Provider>
+);
+
 describe('AvatarGroupItem', () => {
   isConformant({
     Component: AvatarGroupItem,
     displayName: 'AvatarGroupItem',
-    // TODO: enable component-has-static-classnames-object
-    disabledTests: [
-      'component-has-static-classname',
-      'component-has-static-classname-exported',
-      'component-has-static-classnames-object',
-    ],
+    disabledTests: ['component-has-static-classname', 'component-has-static-classname-exported'],
     primarySlot: 'avatar',
+    renderOptions: {
+      wrapper: ContextWrapper,
+    },
+  });
+
+  it('renders only an avatar when context provides true for isOverflow', () => {
+    render(
+      <AvatarGroupContext.Provider value={{ isOverflow: false }}>
+        <AvatarGroupItem data-testid={testId} name="Katri Athokas" />
+      </AvatarGroupContext.Provider>,
+    );
+
+    expect(screen.getByTestId(testId).textContent).toBe('KA');
+  });
+
+  it('renders a label and an avatar when context provides true for isOverflow', () => {
+    render(
+      <AvatarGroupContext.Provider value={{ isOverflow: true }}>
+        <AvatarGroupItem name="Katri Athokas" overflowLabel={<span data-testid={testId}>Test Label</span>} />
+      </AvatarGroupContext.Provider>,
+    );
+
+    expect(screen.getByTestId(testId).textContent).toBe('Test Label');
   });
 
-  // TODO add more tests here, and create visual regression tests in /apps/vr-tests
+  it('sets role to listitem when context provides true for isOverflow', () => {
+    render(
+      <AvatarGroupContext.Provider value={{ isOverflow: true }}>
+        <AvatarGroupItem data-testid={testId} name="Katri Athokas" />
+      </AvatarGroupContext.Provider>,
+    );
 
-  it('renders a default state', () => {
-    const result = render(<AvatarGroupItem name="Default AvatarGroupItem" />);
-    expect(result.container).toMatchSnapshot();
+    expect(screen.getByRole('listitem')).toBeTruthy();
```

Comment:
Not sure if this matters, but would this test be a bit more robust is you asserted against the `.textContent` property here rather than simply existence?

Reply:
Since I'm looking for the role specifically, I'm not sure if asserting against the text content would be desired. I think based on the test title, finding the element with the role `listitem` is sufficient enough. I can update the test to check for the text, but my feeling is that it should stay that way.

Reply:
2 things that come to mind regarding this, one is that `getByRole` is always truthy and will throw an error instead if no item is found. If you want a failed assertion vs a thrown error, then you should use `queryByRole` instead.

Aside from that, I think it might be worth it to both assert that it is truthy and checked the text content after, you can do both in this same test.

Reply:
> Aside from that, I think it might be worth it to both assert that it is truthy and checked the text content after, you can do both in this same test.

Wouldn't `expect(screen.getByRole('listitem').textContent).toBe(...);` handle both in a single assertion? The test will throw if `getByRole()` doesn't find anything so it should always be safe to read `.textContent`.

Reply:
In that case yeah, I think Sean's suggestion make the most sense.